go/vt/vtgate/planbuilder/operators/aggregation_pushing.go

```diff
@@ -57,23 +63,24 @@ func tryPushingDownAggregator(ctx *plancontext.PlanningContext, aggregator *Aggr
 	}
 
 	aggregator.Pushed = true
-	if applyResult != rewrite.SameTree && aggregator.Original {
-		aggregator.aggregateTheAggregates()
-	}
 
 	return
 }
 
 func (a *Aggregator) aggregateTheAggregates() {
-	for i, aggr := range a.Aggregations {
-		// Handle different aggregation operations when pushing down through a sharded route.
-		switch aggr.OpCode {
-		case opcode.AggregateCount, opcode.AggregateCountStar, opcode.AggregateCountDistinct:
-			// All count variations turn into SUM above the Route.
-			// Think of it as we are SUMming together a bunch of distributed COUNTs.
-			aggr.OriginalOpCode, aggr.OpCode = aggr.OpCode, opcode.AggregateSum
-			a.Aggregations[i] = aggr
-		}
+	for i := range a.Aggregations {
+		aggregateTheAggregate(a, i)
+	}
+}
+
+func aggregateTheAggregate(a *Aggregator, i int) {
+	aggr := a.Aggregations[i]
+	switch aggr.OpCode {
+	case opcode.AggregateCount, opcode.AggregateCountStar, opcode.AggregateCountDistinct:
+		// All count variations turn into SUM above the Route.
+		// Think of it as we are SUMming together a bunch of distributed COUNTs.
+		aggr.OriginalOpCode, aggr.OpCode = aggr.OpCode, opcode.AggregateSum
+		a.Aggregations[i] = aggr
 	}
 }
 
```

Comment:
not sure if this split is helpful. probably can be inlined.

Reply:
we are using `aggregateTheAggregate(a *Aggregator, i int)` from other places